### ISSUE TITLE
YAML sort-start comments

### ISSUE THREAD

**Issue by philderbeast** [kitesi/scoped-sort#13]
Thanks for this extension. I'd like to use it with YAML using `sort-start` and `sort-end` comments. Is this possible now or am I making a feature suggestion?

```yaml
packages:
# sort-start --regex /[^\/]/
  - ./pkg-test/
  - ./pkg/
# sort-end
```

**Comment by kitesi** [comment]
TLDR:  This program does not take any language into special consideration (except markdown a little). But since yaml is pretty much just a markdown list from my understanding, it should work if you have reasonable yaml format.

---

Before going into anything, your comments are incorrect. They need to be `{ sort-start --regex /[^\/]/ }` and `{ sort-end }`.

You can also use `--recursive | -r` and make the sort-start top level:

```
#  { sort-start --regex /[^\/]/  --recursive }
packages:
  - ./pkg-test/
  - ./pkg/
more content:
  - ./pkg-two/
  - ./pkg-one/
# { sort-end }
```

This way you don't need to create like 50 sort-comments and can instead try to sort the entire file.

The regex is also not doing anything because it'll match the first blank whitespace and then sort using the text afterwards which is pretty much already what it does. I'm guessing this is just some sample text so a regex might have some value in the real text (not `/[^\/]/` though).

Anyways, I don't have too much experience with yaml, but from my limited knowledge:
if your yaml indents on each sub section it should work.

```
 vars:
    - http_port: 90
    - max_clients: 250
```
But if your labels are on the same indentation as it's properties it won't:
```
  tasks:
  - name: ensure apache is at the latest version
    yum:
      name: httpd
      state: latest
  - name: write the apache config file
    template:
      src: /srv/httpd.j2
      dest: /etc/httpd.conf
    notify:
    - restart apache
  - name: ensure apache is running
```

This is with the assumption you only have a single sort-comment on the whole yaml file.

If you have multiple sort-comments you'll prob be able to get your desired sorts regardless of formatting.

**Comment by kitesi** [comment]
Also might be interested in the [blocksort extension by 1nVitr0](https://github.com/1nVitr0/plugin-vscode-blocksort). Has sort comments and is language aware of yaml, json, etc. Doesn't have `--regex` and some other options though.
